ci-operator/step-registry/kueue-operator/install/catalog/kueue-operator-install-catalog-commands.sh

```diff
@@ -0,0 +1,209 @@
+    redhat_registry_auth=$(echo -n " " "$redhat_auth_user":"$redhat_auth_password" | base64 -w 0)
+
+    stage_registry_path="/var/run/vault/mirror-registry/registry_stage.json"
+    stage_auth_user=$(jq -r '.user' $stage_registry_path)
+    stage_auth_password=$(jq -r '.password' $stage_registry_path)
+    stage_registry_auth=$(echo -n " " "$stage_auth_user":"$stage_auth_password" | base64 -w 0)
+
+    mirror_registry_path="/var/run/vault/mirror-registry/registry_creds"
+    mirror_registry_auth=$(head -n 1 "$mirror_registry_path" | base64 -w 0)
+    
+    echo "Appending the pull secrets to Podman auth configuration file '${XDG_RUNTIME_DIR}/containers/auth.json'..."
+    oc extract secret/pull-secret -n openshift-config --confirm --to ${TMP_DIR}
+    jq --argjson a "{\"registry.redhat.io\": {\"auth\": \"$redhat_registry_auth\"}, \"registry.stage.redhat.io\": {\"auth\": \"$stage_registry_auth\"}, \"${MIRROR_REGISTRY_HOST}\": {\"auth\": \"$mirror_registry_auth\"}}" '.auths |= . + $a' "${TMP_DIR}/.dockerconfigjson" > ${XDG_RUNTIME_DIR}/containers/auth.json
+}
+
+# Applicable for 'disconnected' env
+function install_oc_mirror () {
+    echo "[$(timestamp)] Installing the latest oc-mirror client..."
+    run_command "curl -k -L -o oc-mirror.tar.gz https://mirror.openshift.com/pub/openshift-v4/$(uname -m)/clients/ocp/latest/oc-mirror.tar.gz"
+    run_command "tar -xvzf oc-mirror.tar.gz && chmod +x ./oc-mirror && rm -f oc-mirror.tar.gz"
+}
+
+# Applicable for 'disconnected' env
+function mirror_catalog_and_operator() {
+    # Check if MUST_GATHER_IMAGE is set by the compute step
+    if [[ -z "${MUST_GATHER_IMAGE:-}" ]]; then
+        echo "ERROR: MUST_GATHER_IMAGE not set by compute step"
+        exit 1
+    fi
+
+    echo "Using MUST_GATHER_IMAGE from compute step: ${MUST_GATHER_IMAGE}"
+    
+    # For disconnected environments, the image will be mirrored to the mirror registry
+    # Set the mirrored image path for disconnected environments
+    MIRRORED_MUST_GATHER_IMAGE="${MIRROR_REGISTRY_HOST}/redhat-user-workloads/kueue-operator-tenant/kueue-must-gather"
+    echo "Mirrored MUST_GATHER_IMAGE for disconnected environment: ${MIRRORED_MUST_GATHER_IMAGE}"
+    # Update the environment variable to use the mirrored image for disconnected environments
+    echo "export MUST_GATHER_IMAGE=${MIRRORED_MUST_GATHER_IMAGE}" >> "${SHARED_DIR}/env"
+
+    echo "[$(timestamp)] Creating ImageSetConfiguration for catalog and operator related images..."
+    cat > ${TMP_DIR}/imageset.yaml << EOF
+apiVersion: mirror.openshift.io/v2alpha1
+kind: ImageSetConfiguration
+mirror:
+  operators:
+  - catalog: ${CATSRC_IMG}
+    packages:
+    - name: openshift-cert-manager-operator
+    - name: kueue-operator
+      channels:
+      - name: stable-v1.0
+  additionalImages: # additional images used in e2e test code
+  - name: quay.io/openshift/origin-oauth-proxy:4.14
+  - name: ${MUST_GATHER_IMAGE}
+EOF
```

Comment:
Hi @sohankunkerkar - I am from NetObserv Operator team, this PR was shared on forum-openshift-qe channel, we have a similar use case of mirroring disconnected clusters, would you be willing to make this step (for disconnected clusters) more generic such that other teams like us could also make use of it?

Reply:
Sure, this PR is still in flux. Once it settles down, I’ll try to refactor some of the pieces so other teams can use the same workflow.